core/google/cloud/future/base.py

```diff
@@ -89,15 +92,48 @@ def __init__(self):
         self._done_callbacks = []
 
     @abc.abstractmethod
+    def done(self):
+        """Checks to see if the operation is complete.
+
+        Returns:
+            bool: True if the operation is complete, False otherwise.
+        """
+        # pylint: disable=redundant-returns-doc, missing-raises-doc
+        raise NotImplementedError()
+
+    def running(self):
+        """True if the operation is currently running."""
+        return not self.done()
+
     def _blocking_poll(self, timeout=None):
         """Poll and wait for the Future to be resolved.
 
         Args:
             timeout (int): How long to wait for the operation to complete.
                 If None, wait indefinitely.
         """
-        # pylint: disable=missing-raises
-        raise NotImplementedError()
+        if self._result_set:
+            return
+
+        retry_on = tenacity.retry_if_result(lambda result: result is not True)
+        wait_on = tenacity.wait_exponential(multiplier=1, max=10)
+
+        if timeout is not None:
```

Comment:
How do you feel about using `timeout is None` and swapping the blocks? (You seem to love using `not` even when you not need to.)